azure/datalake/store/multithread.py

```diff
@@ -195,26 +195,42 @@ def clear_saved():
     def hash(self):
         return self._name
 
+
+
     def _setup(self):
         """ Create set of parameters to loop over
         """
-        if "*" not in self.rpath:
-            rfiles = self.client._adlfs.walk(self.rpath, details=True, invalidate_cache=True)
-        else:
+
+        def is_glob_path(path):
+            path = AzureDLPath(path).trim()
+            prefix = path.globless_prefix
+            return not path == prefix
+
```

Comment:
wont it be better to get is_glob_path(self.rpath) once and use that value twice

Reply:
Agreed. Made changes